### DISCUSSION TITLE
Is there a way to give a block of a text and get misspellings position and suggestions ?

### DISCUSSION THREAD

**Discussion by tigitz** [wolfgarbe/symspell#105] [Q&A] [answered]
Couldn't find the answer through the README.md and I'm not familiar with C# so I'm having trouble finding my answer through the source code.

I'm preparing a spellchecker benchmark and I would like to be able to get this type of info:
![image](https://user-images.githubusercontent.com/1524501/107145518-68af2980-6942-11eb-9e0d-933a445d7445.png)

**Comment by wolfgarbe** [comment, marked as answer]
**SymSpell.Lookup()** assumes every input string as single term. 
Lookup (with verbosity set to SymSpell.Verbosity.All) will return all suggestions (within a given maximum-edit distance) ordered by edit-distance, then by word frequency.

**SymSpell.LookupCompound()** supports compound aware automatic spelling correction of multi-word input strings.
It supports compound splitting / decompounding with three cases:
- mistakenly inserted space within a correct word led to two incorrect terms
- mistakenly omitted space between two correct words led to one incorrect combined term
- multiple input terms with/without spelling errors

LookupCompound will return the complete automatically corrected text. There is no specific information returned about which words have been corrected, about their position in the text, nor multiple variants of suggestions per word. It uses Naive Bayes probability for automatically  selecting best word splittings/corrections across the whole text.

SymSpell is an algorithm, rather than a turn-key spelling correction tool. Its purpose is to find all terms/candidates from a large dictionary which are within a maximum edit distance (Damerau-Levenshtein) to an input term in a very short time (to my knowledge much faster than any other algorithm).  The main purpose of the SymSpell library was to provide a sample implementation of the algorithm, to showcase the speed of the algorithm and to allow an easy integration and customization. While SymSpell is not a replacement for a turn-key spelling correction tool, it can be used as core technology to build one if speed is an issue. 

So, the exact functionality you are asking for is currently not implemented in SymSpell. But the SymSpell algorithm/library could easily used/extended to implement it.

**Reply by tigitz** [reply]
Thank you for your substantive reply !

> SymSpell is an algorithm, rather than a turn-key spelling correction tool

And 

>  LookupCompound will return the complete automatically corrected text. There is no specific information returned about which words have been corrected, about their position in the text, nor multiple variants of suggestions per word. It uses Naive Bayes probability for automatically selecting best word splittings/corrections across the whole text.

Definitely helped clear my understanding.

I'll take a look if one of the available ports is providing these extra functionalities.
